sha1solver/sha1solver.c

```diff
@@ -0,0 +1,141 @@
+#include "sha1.h"
+#include "stdio.h"
+#include "string.h"
+#include "stdlib.h"
+
+#define hex(c) ((*(c)>='a')?*(c)-'a'+10:(*(c)>='A')?*(c)-'A'+10:*(c)-'0') 
+
+int SHA1Progress(
+    char *hash_out,
+    char *str,
+    int len,
+    int poz_start,
+    int poz_end,
+    SHA1_CTX * ctx,
+    char *pattern,
+    int pattern_len,
+    int r,
+    int s,
+    int first)
+{
+    unsigned int ii;
+
+    SHA1Init(ctx);
+    for (ii=0; ii<poz_start; ii+=1)
+        SHA1Update(ctx, (const unsigned char*)str + ii, 1);
+
+    SHA1_CTX ctx1;
+    int poz;
+    poz = poz_start;
+    if (!first) {
+	str[poz] ++;
+        if (str[poz] == (const unsigned char) 0) {
+            poz ++;
+        }
+    }
+    while (1) {
```

Comment:
In simple words, when does the loop terminate?

Reply:
I have added comments in the code